src/List.js

```diff
@@ -77,77 +129,79 @@ export function List() {
     let date = new Date();
     const item = items.find((element) => element.id === id);
     const daysSinceLastTransaction = item?.lastPurchasedDate
       ? convertToDays(Math.round(new Date() - item.lastPurchasedDate))
-      : 0;
+      : convertToDays(Math.round(new Date() - item.creationTime));
     const checked = event.target.checked;
     if (checked) {
       const itemRef = doc(db, 'shopping-list', id);
       setDoc(
         itemRef,
         {
           lastPurchasedDate: date.getTime(),
-          previousEstimate: calculateEstimate(
-            item.previousEstimate || parseInt(item.days),
-            daysSinceLastTransaction,
-            item.totalPurchases,
+          previousEstimate: Math.round(
+            calculateEstimate(
+              item.previousEstimate,
+              daysSinceLastTransaction,
+              item.totalPurchases,
+            ),
           ),
           totalPurchases: item.totalPurchases + 1,
         },
         { merge: true },
       );
     }
   };
-  if (items.length) {
-    return (
-      <>
-        <label htmlFor="filterItems">Filter items:</label>
-        <input
-          name="filterItems"
-          type="text"
-          value={filterItem}
-          placeholder="Start typing here..."
-          onChange={(event) => setFilterItem(event.target.value)}
-        ></input>
-        <button onClick={() => setFilterItem('')}>X</button>
-        <ul className="list">
-          {items &&
-            items
-              .filter((item) => !!item.id)
-              .filter((item) =>
-                item.name.toLowerCase().includes(filterItem.toLowerCase()),
-              )
-              .map((item) => {
-                return (
-                  <li key={item.id}>
-                    <input
-                      type="checkbox"
-                      id={`custom-checkbox-${item.id}`}
-                      name={item.name}
-                      value={item.name}
-                      checked={
-                        !!item.lastPurchasedDate &&
-                        new Date() - item.lastPurchasedDate < ONE_MINUTE
-                      }
-                      onChange={(e) => handleChange(item.id, e)}
-                    />
-                    <label htmlFor={`custom-checkbox-${item.id}`}>
-                      {item.name}
-                    </label>
-                  </li>
-                );
-              })}
-        </ul>
-        <NavigationMenu />
-      </>
-    );
-  }
-  return (
+
+  return items.length ? (
+    <>
+      <label htmlFor="filterItems">Filter items:</label>
+      <input
+        name="filterItems"
+        type="text"
+        value={filterItem}
+        placeholder="Start typing here..."
+        onChange={(event) => setFilterItem(event.target.value)}
+      ></input>
+      <button onClick={() => setFilterItem('')}>X</button>
+      <ul className="list">
+        {items &&
+          items
+            .filter((item) => !!item.id)
```

Comment:
Good point! We realized we don't need the first filter anymore as we since resolved the problem it was fixing, so we just deleted the first filter and everything is now working fine!